python/pyspark/cloudpickle.py

```diff
@@ -40,164 +40,126 @@
 NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 """
-
+from __future__ import print_function
 
 import operator
 import os
+import io
 import pickle
 import struct
 import sys
 import types
 from functools import partial
 import itertools
-from copy_reg import _extension_registry, _inverted_registry, _extension_cache
-import new
 import dis
 import traceback
-import platform
-
-PyImp = platform.python_implementation()
-
 
-import logging
-cloudLog = logging.getLogger("Cloud.Transport")
```

Comment:
I'd be interested in chatting more aboutcCloudpickle, but should probably move this discussion to a mailing list since it's hard to link to GitHub line comments.  Mind sending an email to the [dev list](https://spark.apache.org/community.html) and CC'ing me?  My email is `joshrosen@databricks.com`.